plugins/yara/src/plugin.cpp

```diff
@@ -0,0 +1,423 @@
+      case ERROR_TOO_MANY_MATCHES:
+        return caf::make_error(ec::unspecified, "too many matches");
+    }
+    return builder.finish_as_table_slice("yara.match");
+  }
+
+private:
+  static auto callback(YR_SCAN_CONTEXT* context, int message,
+                       void* message_data, void* user_data) -> int {
+    TENZIR_ASSERT(user_data != nullptr);
+    auto* builder = reinterpret_cast<series_builder*>(user_data);
+    if (message == CALLBACK_MSG_RULE_MATCHING) {
+      auto* rule = reinterpret_cast<YR_RULE*>(message_data);
+      TENZIR_DEBUG("got a match for rule {}", rule->identifier);
+      YR_STRING* string = nullptr;
+      yr_rule_strings_foreach(rule, string) {
+        auto row = builder->record();
+        auto rec = row.field("rule").record();
+        rec.field("identifier").data(rule->identifier);
+        rec.field("namespace").data(std::string_view{rule->ns->name});
+        auto rule_string
+          = std::string_view{reinterpret_cast<const char*>(string->string),
+                             detail::narrow_cast<size_t>(string->length)};
+        rec.field("string").data(rule_string);
+        const char* tag = nullptr;
+        auto tags = rec.field("tags").list();
+        yr_rule_tags_foreach(rule, tag) {
+          tags.data(std::string_view{tag});
+        }
+        auto meta_rec = rec.field("meta").record();
+        YR_META* meta = nullptr;
+        yr_rule_metas_foreach(rule, meta) {
+          auto identifier = std::string_view{meta->identifier};
+          if (meta->type == META_TYPE_INTEGER)
+            meta_rec.field(identifier).data(int64_t{meta->integer});
+          else if (meta->type == META_TYPE_BOOLEAN)
+            meta_rec.field(identifier).data(meta->integer != 0);
+          else
+            meta_rec.field(identifier).data(std::string_view{meta->string});
+        }
+        YR_MATCH* match = nullptr;
+        auto list = rec.field("matches").list();
+        yr_string_matches_foreach(context, string, match) {
+          auto match_rec = list.record();
+          match_rec.field("identifier").data(string->identifier);
+          auto bytes = std::span<const std::byte>{
+            reinterpret_cast<const std::byte*>(match->data),
+            detail::narrow_cast<size_t>(match->data_length)};
+          // TODO: switch to blob type once available.
+          auto str = std::string_view{
+            reinterpret_cast<const char*>(bytes.data()), bytes.size()};
+          match_rec.field("data").data(str);
+          match_rec.field("base").data(match->base);
+          match_rec.field("offset").data(match->offset);
+          match_rec.field("match_length")
+            .data(detail::narrow_cast<uint64_t>(match->match_length));
+          // TODO: Once we can upgrade to newer versions of libyara, uncomment
+          // the line below. YR_MATCH::xor_key is not available in the version
+          // we get on Debian.
+          // match_rec.field("xor_key").data(uint64_t{match->xor_key});
```

Comment:
You can also do this with if-constexpr-requires-requires:

```cpp
if constexpr (requires requires { match->xor_key; }) {
  // ...
}
```

Reply:
Neat. Does that work with CI? I.e., are we building it exactly on one platform?

Reply:
Ah, it may be problematic for the integration tests, so it would probably require a `drop xor_key` in them if you're on Debian. I forgot about that.

Not sure how important this value is, feel free to just ignore this and leave the TODO as-is.

Reply:
Yeah, let's leave the TODO in there until we get a user feedback.